src/Node.ts

```diff
@@ -172,11 +183,11 @@ export abstract class Node<Config extends NodeConfig = NodeConfig> {
     // all change event listeners are attached to the prototype
   }
 
-  hasChildren() {
+  hasChildren(): boolean {
```

Comment:
Personally, I don't like having return types everywhere. I prefer less code as it is more clean. 

> I've explicitly typed the return value of many functions that previously had their return types inferred. This provides insurance against unintentional changes to return types.

It is a good point. But still I prefer not to write type. Probably is is a good idea to write them on complex functions with many returns. But on small one, with one return, most likely we don't really need it. Especially if it is just one direct value return.